astropy/units/format/cds.py

```diff
@@ -189,57 +183,34 @@ def p_unit_expression(p):
 
         def p_factor(p):
             """
-            factor : signed_float X UINT SIGN UINT
-                   | UINT X UINT SIGN UINT
-                   | UINT SIGN UINT
-                   | UINT
-                   | signed_float
+            factor : FLOAT X INT INT
```

Comment:
Had to think a bit about it, but of course `INT INT` is equivalent to requiring a sign in between, otherwise it would be a single int.